docs/src/lang/functions.md

```diff
@@ -231,15 +228,24 @@ that are computed under such a binding.
     \* to a function from 1..n to n + i. Like an array of arrays.
 ```
 
-**Example in Python:** TLA+ functions are immutable, so we are using [frozendict]:
+**Example in Python:**
 
 ```python
-  X = frozenset({ 1, 2, 3 })
-  frozendict({ x: 2 * x for x in X })
-  frozendict({ (x, y): x * y for x in X for y in X })
-  Y = frozenset({ 4, 5, 6 })
-  XY = frozenset((x, y) for x in X for y in Y)
-  frozendict({ (x, y): x + y  for (x, y) in XY })
+# TLA: [ x \in 1..3 |-> 2 * x ]
+{x: 2 * x for x in range(1, 4)}
+# TLA: [ x, y \in 1..3 |-> x * y ]
+{(x, y): x * y for x in range(1, 4) for y in range(1, 4)}
+# TLA: [ <<x, y>> \in (1..3) \X (4..6) |-> x + y ]
+xy = {(x, y) for x in range(1, 4) for y in range(4, 7)}
+{(x, y): x + y for (x, y) in xy}
+# TLA: [ n \in 1..3 |->
+#        [ i \in 1..n |-> n + i ]]
+# We have to use frozenset to use a set as key in a dict.
+{
+    n: {frozenset(range(1, i + 1)): n + 1}
+    for n in range(1, 4)
+    for i in range(1, n + 1)
+}
```

Comment:
Are you sure that the python code reflects the TLA+ expression? From what I see in python repl, the result of evaluating the python expression is:

```
{1: {frozenset({1}): 2}, 2: {frozenset({1, 2}): 3}, 3: {frozenset({1, 2, 3}): 4}}
```

The TLA code is constructing a function of `n \in 1..3` that returns a function of `i \in 1..n` to `n+i`.

Reply:
Right! So it would be

```python
{
    n: {i: n + i for i in range(1, n + 1)}
    for n in range(1, 4)
}
```

That's nice because in this case there is no need for any `frozenset` or `frozendict` in this document.

Reply:
fixed